Add Documentation file to the Common Folder

hamartia-mk opened the pull request:
To understand the codebase.
1. Created the doc file
2. Added a table of contents.
3. Wrote up documentation for the Bid and Card class components

[Branch: documentation-patch]

hamartia-mk committed:
Add Documentation file to the Common Folder

To understand the codebase.
1. Created the doc file
2. Added a table of contents.
3. Wrote up documentation for the Bid and Card class components

[Branch: documentation-patch]

Hamartia666 commented:
I'm not sure what is the best location for the documentation, I currently put this in the same folder as the code.
Perhaps I should make a separate directory, for example `Internal Docs`?

michalklisch commented:
I'd say having a dedicated folder for docs is better. Easier to keep things clean.

hamartia-mk committed:
Created Documentation Directory

Additionally, moved commonfiles docs to the Documentation Directory
For clarity and structure

[Branch: documentation-patch]
